src/api/resourses/users/UserController.js

```diff
@@ -39,18 +27,17 @@ class UserController {
   }
 
   static login(req, res) {
-    const queryString = "SELECT * FROM users WHERE email = $1";
-    //  Database query
+    const queryString = "SELECT  id,email,username,password,isadmin FROM users WHERE email = $1 ";
     db.query(queryString, [req.body.email])
       .then(data => {
         const user = data.rows[0];
         if (!user) {
           res.status(404).json({ status: 404, error: "User not Found" });
         }
-        if (!bcrypt.compareSync(req.body.password, user.password)) {
+        if (!checkPassword(req.body.password, user.password)) {
           res.status(404).json({ status: 404, error: "invalid credentials" });
         }
-        const token = createToken(req.body.email);
+        const token = createToken(user.id, user.username, user.isadmin);
         return res
```

Comment:
Why do you need to add the username when creating the token

Reply:
The username is passed along to the req object after verification

Reply:
You don't need the username, the user id can do all that, with the id, you can identify any user because it is a unique value.

Reply:
ok what I felt is that if I ever need username in my application I don't need to query database again

Reply:
I will remove the username from the token generator

Reply:
username removed